src/H5Tconv.c

```diff
@@ -185,27 +177,47 @@ H5T_get_force_conv(const H5T_t *dt)
  *-------------------------------------------------------------------------
  */
 herr_t
-H5T__reverse_order(uint8_t *rev, uint8_t *s, size_t size, H5T_order_t order)
+H5T__reverse_order(uint8_t *rev, uint8_t *s, const H5T_t *dtype)
```

Comment:
After looking at this again, the function name is very misleading.  It should be "H5T__to_le_order" or similar.  If you are on a little-endian machine, it definitely doesn't reverse the order.

Reply:
Agreed and ideally we would be able to refactor the conversion functions to not have to byte-swap data into little-endian order in the first place, removing the need for reversing the buffer before calling the exception callback function.